src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -102,6 +100,11 @@ protected void parseCreateField(ParseContext context, int dimension) throws IOEx
         if (fieldType.stored()) {
             context.doc().add(new StoredField(name(), point.toString()));
         }
+
+        if (docValuesFieldType != null) {
+            context.doc().add(new VectorField(name(), array, docValuesFieldType));
+        }
```

Comment:
Why not move the building of docvalue closer to required place? 
```suggestion
        if (this.hasDocValues) {
            var docValueField = buildDocValuesFieldType(this.knnMethod.getKnnEngine())
            context.doc().add(new VectorField(name(), array, docValueField));
        }
```

Reply:
Main reason is to avoid building for each document.